index.html

```diff
@@ -1,12 +1,13 @@
 <!DOCTYPE html>
 <html lang="en">
 <head>
+	<base href="<%= htmlWebpackPlugin.options.baseUrl %>" />
```

Comment:
I did not yet get why we need this

Reply:
@tempelgogo

Reply:
The [base](https://developer.mozilla.org/de/docs/Web/HTML/Element/base) is a relict from a time when HTML was used to build websites, not Web Applications.
It enable's relative urls in our web document, by prefixing them with the base url.
This case especially matters on deep links in history mode, 'cause they do not reflect the real path. Take these paths as example, imagine we are re-loading the page:
- `http://fe.serv3r.cloud:8300/files/list/home`
- `http://fe.serv3r.cloud:8300/files`

Fulfilling the requirement to deploy phoenix in a subdirectory makes this even worse:
- `http://fe.serv3r.cloud:8300/phoenix/files`
- `http://fe.serv3r.cloud:8300/demos/vue/phoenix/master-branch/files`

Everything beginning at `files/` is getting rewritten to `index.html` with corresponding params, we don't know which part of the url is *real*, apps don't know their *deployment base path* like `http://fe.serv3r.cloud:8300/demos/vue/phoenix/master-branch/files.bundle.js`

Where to find the index bundle ?

Webpack as our script bundler is basically able to **traverse through the import/require dependency tree**, which is awesome to tree-shake code duplicates away or adjust import paths.
Unfortunately our webpack isn't able to do so, due to our dynamic require.js-based app loading system.  
There isn't even a guaranty that all apps are available while bundling… so how to modify them ?

This problem has already occurred in many places such as cross-app linkage, including the polyfills, …  
I am not aware of any other **least-effort** solution, which fulfills **all our current requirements**, especially that apps must support dynamic installation.

JavaScript was never made for this funky stuff…

Reply:
The `[html-webpack-plugin](https://github.com/jantimon/html-webpack-plugin#options)` (which we are currently using) has support for an *base* attribute.
Unfortunately I couldn't convince it to work. So I've injected it old-school styled manually... which has some disadvantages, because webpack may put code before it, which we do't want it to; *base* url has to be set before it's used first.
Finally it's performing better as before.

Reply:
So this means that after deployment the index.html has to be manipulated.
No Idea If this is acceptable....